File: EMTmerci2/Assets/Scripts/MainCharacter/Movement/Merci_Movement.cs
//using UnityEngine;
//using System.Collections;
//
//
//enum aiState{IdleStill, RunningTo, Hiding, Operating, Dead
//}
//
//
//
//
//public class Merci_Movement : MonoBehaviour {
//	
////InvokeRepeating("StateLogic", 0.0, 0.05);
//	
//	
//	
//public aiState state;
//
//	void Awake(){
//	
//		animator = GetComponent<Animator;
//	
//}
//	
//	
//public Transform target;
//public bool HasPatient;
//public bool AtTarget;
//public bool TakingCover;
//
////RunningTo variables
//public float RunSpeed = 1;
//
//
//private Animator animator;
//	
//	
//	
//	// Use this for initialization
////	void Start () {
////	
////		yield StateMachine();
////		
////	}
////	
////	void StateLogic(){
////		
////		if(Dead == true){
////			
////			state = aiState.Dead;
////			
////		} else
////		
////		if((HasPatient == true) && (AtTarget == true)){
////			
////			state = aiState.Operating;
////			
////			
////		}else if(TakingCover == true){ 
////		
////			state = aiState.Hiding;
////			
////			} else if(AtTarget == true){
////				
////					state = aiState.IdleStill;
////				
////				} else if(HasPatient = true){
////					
////					state = aiState.RunningTo;
////				}
////				
////			Debug.Log(state);
////		
////			}
////	
////	void StateMachine(){
////		while(true){
////			switch(state){
////				
//////				case aiState.Dead;
//////				Dead();
//////				break;
//////				
//////				case aiState.Operating;
//////				Operating();
//////				break;
//////				
//////				case aiState.Hiding;
//////				Hiding();
//////				break;
//////				
//////				case aiState.IdleStill;
//////				Idle();
//////				break;
//////				
//////				case aiState.RunningTo;
//////		
//////		
//////				}
//////			yield
//////			}
//////	
//////
//////		
//////	void Dead(){
////			
////			
////		}
////		
////		
////	void Operating(){
////			
////			
////		}
////		
////		
////	void Hiding(){
////			
////			
////		}
////		
////		
////	void IdleStill(){
////			
////			
////		}
////		
////	void RunningTo(){
////		
////			
////			public CharacterController controller = GetComponent(CharacterController);
////			
////			controller.SimpleMove(transform.TransformDirection(Vector3.forward * RunSpeed));
////			
//
//}

File: EMTmerci2/Assets/Scripts/MainCharacter/Movement/Merci_Movement.js
#pragma strict

var Dead : boolean;
var target : Transform;
var HasPatient : boolean;
var AtTarget : boolean;
var TakingCover : boolean;
var AnimatingBones : Transform;

var RunSpeed : float = 5;
//var animator : Animator;

//visibility items variables
var GunOut : Transform;
var GunBack : Transform;
var LazerOut : Transform;
var Lazer2Out : Transform;
var SawOut : Transform;

enum merciState{IdleStill, RunningTo, Hiding, Operating, isDead
}
var state : merciState;

InvokeRepeating("StateLogic", 0, 0.05);

function Start () {

	yield StateMachine();

}

function StateLogic(){

		if(Dead == true){
			
			state = merciState.isDead;
			
		} else
		
		if((HasPatient == true) && (AtTarget == true)){
			
			state = merciState.Operating;
			
			
		}else if(TakingCover == true){ 
		
			state = merciState.Hiding;
			
			} else if(HasPatient == false){
				
					state = merciState.IdleStill;
				
				} else{
					
					state = merciState.RunningTo;
				}
				
//			Debug.Log(state);
	

}


function StateMachine(){
		while(true){
			switch(state){
				
				case merciState.isDead:
				isDead();
				break;
				
				case merciState.Operating:
				Operate();
				break;
				
				case merciState.Hiding:
				Hide();
				break;
				
				case merciState.IdleStill:
				Idle();
				break;
				
				case merciState.RunningTo:
				Running();
				break;
					}
				yield;
	
			}
			
				
			
			
			
			

}
	function isDead(){
			
			
		}
		
		
	function Operate(){
			
			AnimatingBones.SendMessage("Operate");
			GunBack.renderer.enabled = false;
			GunOut.renderer.enabled = true;
			LazerOut.renderer.enabled = true;
			Lazer2Out.renderer.enabled = true;
			SawOut.renderer.enabled = true;
			
		}
		
		
	function Hide(){
			
			
		}
		
		
	function Idle(){
			
			AnimatingBones.SendMessage("Idle");
			
		}
		
	function Running(){
		
		var controller : CharacterController = GetComponent(CharacterController);
		var RotationSpeed : float = 10;
		var NeededRotation = Quaternion.LookRotation(target.position - transform.position);
		
		
			transform.rotation = Quaternion.Slerp(transform.rotation, NeededRotation, Time.deltaTime * RotationSpeed);
			transform.rotation.x = 0;
			transform.rotation.z = 0;
			AnimatingBones.SendMessage("Run");
			GunBack.renderer.enabled = true;
			GunOut.renderer.enabled = false;
			LazerOut.renderer.enabled = false;
			Lazer2Out.renderer.enabled = false;
			SawOut.renderer.enabled = false;
			
			controller.SimpleMove(transform.TransformDirection(Vector3.forward * RunSpeed));
			

}


//Behavorial Interactions Below//

	function OnTriggerEnter(hit : Collider){
	
		if(hit.gameObject.transform == target){
		
		Camera.main.SendMessage("Operate");
		AtTarget = true;
		hit.gameObject.collider.enabled = false;
		
		
		}
	
	}


	function CubeHere(NewTarget){
	
		if(NewTarget != target){
			target.SendMessage("OnTriggerEnter");
			target.gameObject.collider.enabled = true;
			target = NewTarget;
			AtTarget = false;
			HasPatient = true;
		}
	}
	
	function Slow(){
	
		RunSpeed -= 10;
	}
	
	function Fast(){
	
		RunSpeed += 10;
	}
@script RequireComponent(Collider)